### ISSUE TITLE
FR Heading Search

### ISSUE THREAD

**Issue by chrisgrieser** [tadashi-aikawa/obsidian-another-quick-switcher#28]
having the ability to search through headings of the current note is a feature that I was looking for a few times. 

It feels to me like this plugin might be a good match, since it works similarly to backup search. 

This would allow you to quickly navigate big notes with lots of headings. While you can do that with the outline core plugin as well, that plugin cannot be navigated solely with the keyboard.

**Comment by tadashi-aikawa** [comment]
Hi, @chrisgrieser :)

I use that feature with Quick Switcher++, `Open in Headings Mode`. However, if there are advantages to implementing the same or advanced feature to Another Quick Switcher, I'll consider it 🤔

**Comment by chrisgrieser** [comment]
oh, wasn't aware that the plugin has what I was looking for 🙂 

Well, in that case you of course don't need to simply replicate an existing feature

**Comment by FelipeRearden** [comment]
Hello @tadashi-aikawa and @chrisgrieser !!!!

Forgive for step into yours FR. Since I know QS++ pretty well, I think I can add some information about it. 

What @chrisgrieser is asking in this FR is about a modal that shows all headings from the **Active File**. That's it, that the FR. 
- For example id the active note is like this 

```
Cities of USA

## East Cost
text here ...

### New York
text here ...

### Boston
text here ...

## West Coast
text here ...

### San Jose
text here ...
```

The new command would show a Modal showing 5 lines with each header from the note:

```
H2 East Cost
H3 New York
H3 Boston
H2 West Coast
H3 San Jose
```
QS++ does not have this feature. QS++ has two features related to this :)

### Feature 1: `Open in Headings Mode` 

Let's the user select ALL FILES form the vault using the Headers as the search criteria

For Example if Note A and Note B have the heading `## Michael Jordan`, it will show these two Notes when using the command

![image](https://user-images.githubusercontent.com/85180696/161423807-2cf9201b-7fa9-43c2-8ae5-03531549b801.jpeg)


### Feature 2: `Open in Symbol Mode` 

This feature let the user navigate to many things only in the ACTIVE NOTE. 

It opens a modal with information from the active note. The user can choose what will be showed in the modal:

![image](https://user-images.githubusercontent.com/85180696/161424007-c8f5fc97-4467-4a7c-b035-4e8a20657768.jpeg)

As you can see it has an option to show the Headings of the active file. But to achieve this we lost the option of select other types of content of the note 😢

### A Good Feature for AQS !!!!!

Would be amazing for my workflow if AQS bring a feature to show a modal with only the Headings of the Active Note. This way we have all of the world of workflows covered 🙏 🙏 🙏

--- 

Sorry for the long text. Let me know if something is not clear 

Thanks for reading this. 

Have a great day!!!!

**Comment by tadashi-aikawa** [comment]
@FelipeRearden 
First, thank you for your follow-up!.

My post was wrong, not `Open in Headings Mode` but `Open in Symbol Mode` 😅 

> As you can see it has an option to show the Headings of the active file. But to achieve this we lost the option of select other types of content of the note 😢

Right. It seems valuable to implement the feature to Another Quick Switcher. So I'll reopen this issue and try to implement it 👍

**Comment by FelipeRearden** [comment]
> First, thank you for your follow-up!.

I'm glad to help 🙃

> My post was wrong, not `Open in Headings Mode` but `Open in Symbol Mode` 😅 

😀 

> Right. It seems valuable to implement the feature to Another Quick Switcher. So I'll reopen this issue and try to implement it 👍 

Thank you very much for trying to implement it !!!!!!!!!!!!!

I'm so happy !!!!!!

Have a great day 🌞

**Comment by tadashi-aikawa** [comment]
@chrisgrieser @FelipeRearden 
Coming soon 😉 

https://user-images.githubusercontent.com/9500018/166876741-58412325-bba2-4a66-829c-883b8fd2f433.mp4

`Header search in file` in Another Quick Switcher is different from other plugins.

- Show all headers even after filtering to retain file structures in the brain
- Jump to the first hit suggestion automatically and move next/previous by `Tab/Shift+Tab`

I don't know whether users prefer it or not (Of course, I like it!). I want to see a little and might add an option later if it needs.

**Comment by chrisgrieser** [comment]
👁️👄👁️

**Comment by FelipeRearden** [comment]
Looks AMAZING @tadashi-aikawa !!!!!!

👀 👀 👀

**Comment by tadashi-aikawa** [comment]
OK. I'll release 👍 

![2022-05-05](https://user-images.githubusercontent.com/9500018/166905283-d376aae8-7d44-40bd-8d57-89fc4efcfbd0.gif)

**Comment by tadashi-aikawa** [comment]
@chrisgrieser @FelipeRearden 
Released [v4.10.0](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.10.0) 🚀

**Comment by FelipeRearden** [comment]
> @chrisgrieser @FelipeRearden 
> 
> Released [v4.10.0](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.10.0) 🚀 
> 
> 

Thank you @tadashi-aikawa !!! 

I'm starting my tests and I get back to you in less than an hour :)

**Comment by FelipeRearden** [comment]
Hello @tadashi-aikawa !!!!

It’s absolutely amazing to see this Feature. I’m bring some ideas based on my workflow for us to talk about 

🖥 My Environment -> Obsidian **Mobile** iPadOS with mouse and keyboard

# Ideas 💡

## 01. Open the Modal Focused in the nearby Header (based caret cursor)

**Example** 

![59ACC410-FCD6-4AB4-9E54-07A632C95454](https://user-images.githubusercontent.com/85180696/166915258-2c862b51-f9fb-4a69-8aa4-3df97450eb48.jpeg)

This  note has 3 headers: `NBA` , `Chicago Bulls` and `Michael Jordan `

The caret cursor is active in a line inside the header `Chicago Bulls` 

When we open the modal, it is focused in the first header  `NBA`

💡 My idea is to focus the modal in the header that belongs the caret cursors -> in this case `Chicago Bulls` 

This would help us to quick navigate inside long notes with many and multiple headers (and H3 that has 15 H4 inside) . 

This is the last header of a note with 19 H2 and at least 5 H3 inside each H2 
![E1B82487-852C-45C4-9D4A-6942A80C6593](https://user-images.githubusercontent.com/85180696/166915819-c98711d4-7cbb-4173-8642-dbf95aec585d.png)

## 02. Open Header in the Top of the Note

Would be amazing if after selecting a header from the modal, the note could **jump** to the top of the screen. This only happens with long notes:

Step 1: Open a Long Note
Step 2: Open Modal
Step 3: Select a Header from the end of the note 
Step 4: Look where the header are in the screen

![36604B13-5F59-4706-94D1-F518A334EB9C](https://user-images.githubusercontent.com/85180696/166916862-993dd9a3-015c-40a7-af24-d1a23b072484.jpeg)

![E8BBFB61-4705-4531-AB77-4AC5DE1EB895](https://user-images.githubusercontent.com/85180696/166916899-432627f9-b0da-468c-a91e-01d065e08399.jpeg)

If possible,  would be great if the header opens in the top of the note 🙏 
![0BA9AB8B-ED7B-4490-9E42-D6EC1B516F92](https://user-images.githubusercontent.com/85180696/166917138-335dc346-bcf7-4a31-977d-1faa8ec1d2a1.jpeg)

# Question ❓

## Support  for Reading

Would be possible to add support for reading Mode ????? 

Would be amazing to navigate between headers while reading a long note in Reading Mode 

---

I just wanna say thank you for one more amazing feature. 

Let me know what you think about this ideas :)

I gonna keep playing and I let you know if I found something to share 

Have a great day ☀️

**Comment by tadashi-aikawa** [comment]
@FelipeRearden 
Thank you for your kind report :)

> 01. Open the Modal Focused in the nearby Header (based caret cursor)

Oh, it's cool! I have never noticed it because I don't have written many headers. I can probably implement it, but it seems not easy to scroll the current header center.

> 02. Open Header in the Top of the Note

It doesn't jump to the top of the screen. However, wouldn't it jump to the center of the screen? I confirmed it (jump to center) on my devices, Windows PC, Android Phone, and iPad Pro.

> Support for Reading

Thanks! That is my lack of consideration😅. I don't know whether it is possible or not, but I'll try it 👍

**Comment by chrisgrieser** [comment]
First of all, wow, this feature is way better than I could have imagined! 🥳 
With the way this plugin works, I am now considering to turn of the outline core plugin.

---

> - Jump to the first hit suggestion automatically and move next/previous by Tab/Shift+Tab

Could there be an option to switch this around, e.g. make `tab` move to the next item, and make `down`/`cmd/ctrl+j` move to the next hit? Could be subjective, but I think this way would be more intuitive for me.

---

Also, I support both of @FelipeRearden's requests. There would add a lot of polish to this already amazing feature.

**Comment by tadashi-aikawa** [comment]
FYI: I think it's improved a bit.

- Support for Reading
    - Done (Same level as the QS++)
- Open Header in the Top of the Note
    - Reading mode is OK.
    - Editor mode is not, but same level as the QS++
- Open the Modal Focused in the nearby Header (based caret cursor)
    - In progress
- make tab move to the next item, and make down/cmd/ctrl+j move to the next hit?
    - In progress

**Comment by FelipeRearden** [comment]
> Also, I support both of @FelipeRearden's requests. There would add a lot of polish to this already amazing feature.

Thanks for feedback @chrisgrieser  :) 

@tadashi-aikawa  once again is bringing to us a 💎

**Comment by FelipeRearden** [comment]
> I could reproduce it. In my case, for now, it seems impossible to fix it because it occurs in native jump to line command, too. 😓

> But, QS++ works correctly... 🤔 (any chance?)

In my case QS++ works for NoteA and don’t work for NoteB works for NoteC and don’t work for NoteD (I don’t know the english word to say that is just working in some cases)

> FYI: I think it's improved a bit.
> Support for Reading
> Done (Same level as the QS++)
> Open Header in the Top of the Note
> Reading mode is OK.
> Editor mode is not, but same level as the QS++
>Open the Modal Focused in the nearby Header (based caret cursor)
>In progress
> make tab move to the next item, and make down/cmd/ctrl+j move to the next hit?
>In progress

👀 ❤️ 🚀 🙏 🎉

You area legend @tadashi-aikawa :)

**Comment by tyf2018** [comment]
The new function is great, and it would be even better if can jump to the corresponding position in the note after opening the header in reading mode.

**Comment by tadashi-aikawa** [comment]
@chrisgrieser @FelipeRearden @tyf2018 
I have released [4.11.0-beta3](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.11.0-beta3) 🚀 

Please confirm it via BRAT 🙏

**Comment by tyf2018** [comment]
> 请通过 BRAT 确认 🙏

Great, read mode works fine too. 
Your efficiency is amazing!

**Comment by chrisgrieser** [comment]
- Reading Mode works
- Scrolling works, too 👍

**Comment by tadashi-aikawa** [comment]
> In my case QS++ works for NoteA and don’t work for NoteB works for NoteC and don’t work for NoteD (I don’t know the english word to say that is just working in some cases)

@FelipeRearden 
How about the above case?

**Comment by tadashi-aikawa** [comment]
@FelipeRearden 
Thanks! That's great 😄

**Comment by tadashi-aikawa** [comment]
`Eliminate unnecessary decorations as headers`

https://user-images.githubusercontent.com/9500018/167246455-5a95b1e3-923b-4a81-9253-ce341797f24b.mp4

**Comment by FelipeRearden** [comment]
> `Eliminate unnecessary decorations as headers`

Fantastic Idea @tadashi-aikawa !!!!!!!!!!!

👀 would be great if it could eliminate html tags like in this header …

`###### Videos <s id=rr>Lorem Ipsum</s> `

**Comment by tadashi-aikawa** [comment]
> Open the Modal Focused in the nearby Header (based caret cursor)

Maybe, OK. It was more complicated than I could have imagined, but works 😅 

https://user-images.githubusercontent.com/9500018/167249445-e29d10b1-72e9-44e4-afc7-ff7cd479347b.mp4

**Comment by FelipeRearden** [comment]
> > Open the Modal Focused in the nearby Header (based caret cursor)
> 
> 
> 
> Maybe, OK. It was more complicated than I could have imagined, but works 😅 

🎉 🎉 🎉 Absolutely fantastic @tadashi-aikawa !!!!!!!!

Thank you very much for this feature 💙

**Comment by tadashi-aikawa** [comment]
Only one left

> - make tab move to the next item, and make down/cmd/ctrl+j move to the next hit?

**Comment by FelipeRearden** [comment]
@tadashi-aikawa !!!!

I have one more idea about this FR to share with everybody

It's very simple to explain -> **If a header is FOLD, when we select a Header from the Modal, it will open it !**

## My Workflow

I have a bunch of notes that I call cheatsheet for a quick access of the documentation from every workflow that I have in my vault

🎥 A Example from my sources capture workflow
![video](https://user-images.githubusercontent.com/85180696/167251753-cd520a17-0d36-4acd-b86b-0285c6328838.MOV)

Each one of the headers are folded when I open the Note

My Idea is after selecting `Video` from the Modal, AQS will open the folded Header!

I don't know if is feasible but would be great :)

---

🆘 On `4.10.0` When We have a H3 inside a H2, it will open H2 when I select an H3

🎥 Take a look Open Chicago Bulls (H3) opens NBA (H2)
![video](https://user-images.githubusercontent.com/85180696/167251774-e7ab132e-6023-455f-86fe-c140883685d4.MOV)

---

Forgive for not having this insight before. I was about to open this note and I have the idea :)

**Comment by tadashi-aikawa** [comment]
@chrisgrieser @FelipeRearden @tyf2018 
I have released [4.11.0-beta4](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.11.0-beta4) 🚀 

- It includes all implementations about this issue 🦾 (except for folding issue)
- If there are no critical problems, I'll release it tomorrow officially

@FelipeRearden 
 I'll try to implement your latest feature request in the next version 👍

**Comment by tadashi-aikawa** [comment]
@chrisgrieser 
You can customize keys here :)

![image](https://user-images.githubusercontent.com/9500018/167254132-aafd6ba5-8654-4a69-87d6-ffce036c0f6f.png)

**Comment by FelipeRearden** [comment]
> @chrisgrieser @FelipeRearden @tyf2018 
> 
> I have released [4.11.0-beta4](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.11.0-beta4) 🚀 

Thank you very much @tadashi-aikawa !!!!

I'm starting tests right now and I get to you in less than an hour 🚀

> @FelipeRearden 
> 
>  I'll try to implement your latest feature request in the next version 👍 

Thank you very much for considering this new feature and for trying to implement 🍀

**Comment by chrisgrieser** [comment]
fantastic, works great, thanks! 🤩 

I tried it out and I noticed that `move to next/previous hit` doesn't do anything when I didn't enter at least one letter as query? Is this intended?

Because I can also see the heading search to be useful to quickly switcher to a heading when you only have a few. And in *that* case, you might simply wanna navigate down, without entering a search query. But then it is quite counterintuitive to me to switch to the `navigate` keys, since I would still use the `move to next/previous hit` keys as they are the standard keys to use.

Long story short, I would suggest a minor change: when the user has entered *no* query at all (= the input field is empty), `move to next/previous hit` and `navigate` should both work the same – all items are "hits" in that case.

**Comment by FelipeRearden** [comment]
> Long story short, I would suggest a minor change: when the user has entered *no* query at all (= the input field is empty), `move to next/previous hit` and `navigate` should both work the same – all items are "hits" in that case.

In my humble opinion, it's a spectacular idea @chrisgrieser !

**Comment by FelipeRearden** [comment]
> > I have released [4.11.0-beta4](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.11.0-beta4) 🚀 

## My Environment

A brand new vault on Obsidian 0.14.6
the same long note that I have been using for the stress test

ℹ️ After the public release I will redo the tests on iPadOS

## Test Items 

1️⃣ **Respect a header position both after opening dialog and after filtering (#28)**

✅ Live Preview
✅ Source Mode

2️⃣ Eliminate unnecessary decorations as headers (#28)

✅ Reading Mode
✅ Live Preview
✅ Source Mode

3️⃣ Enable to change hotkeys about navigate and move to hit in the "header search in file" dialog  (#28)

✅ Reading Mode
✅ Live Preview
✅ Source Mode

---

Working perfect on Version `4.11.0-beta4` on Obsidian 0.14.6

**Comment by tadashi-aikawa** [comment]
@chrisgrieser
Thanks for your quick feedbacks! 

> I tried it out and I noticed that move to next/previous hit doesn't do anything when I didn't enter at least one letter as query? Is this intended?

Yes. This feature was intended to move search results.

> I would suggest a minor change: when the user has entered no query at all (= the input field is empty), move to next/previous hit and navigate should both work the same – all items are "hits" in that case.

However, your suggestion would be helpful for some users, and not much of disadvantages. So I'll implement it 👍

**Comment by tadashi-aikawa** [comment]
@FelipeRearden 
Thank you for your quick feedback and tests!

Your work is very helpful because I don't have enough times to verify all supported OS and modes ☺️

**Comment by tadashi-aikawa** [comment]
@chrisgrieser @FelipeRearden 
Released [v4.11.0-beta5](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.11.0-beta5) 🚀

**Comment by FelipeRearden** [comment]
> @FelipeRearden 
> 
> Thank you for your quick feedback and tests!
> 
> Your work is very helpful because I don't have enough times to verify all supported OS and modes ☺️ 

@tadashi-aikawa thank you very much for this comment 😀

I'm extremely happy to know that my tests are helpful. Means a lot to me 😁

You can always count on me to do the tests. I will work hard on every beta release to tests many scenarios as possible 💙

**Comment by FelipeRearden** [comment]
> Released [v4.11.0-beta5](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.11.0-beta5) 🚀 

Starting tests now 🚀 I get back in a few minutes .....

**Comment by FelipeRearden** [comment]
> Released [v4.11.0-beta5](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.11.0-beta5) 🚀

Let's wait for @chrisgrieser  feedback but If I understood correctly, working perfect on version `4.11.0-beta5` on Obsidian `0.14.6` 

**Details below** 🔽

*I set all hotkeys to one of the two options and test the behavior between `EMPTY/NOT_EMPTY` on LP, READING and SOURCE MODE*

## Scenario 1: All Hotkeys = `navigate`

🧮 **Expected Results**

ℹ️ Empty Text field = `navigate`

| Hotkey        | Live Preview | Reading | Source |
|:--------------|:-------------|:--------|:-------|
| CRTL+J/K      |  ✅          | ✅      | ✅     |
| CRTL+N/P      |  ✅          | ✅      | ✅     |
| Arrow UP/DOWN |  ✅          | ✅      | ✅     |
| TAB/Shift+TAB |  ✅          | ✅      | ✅     |  


ℹ️ Not Empty Text filed = `navigate`

| Hotkey        | Live Preview | Reading | Source |
|:--------------|:-------------|:--------|:-------|
| CRTL+J/K      |  ✅          | ✅      | ✅     |
| CRTL+N/P      |  ✅          | ✅      | ✅     |
| Arrow UP/DOWN |  ✅          | ✅      | ✅     |
| TAB/Shift+TAB |  ✅          | ✅      | ✅     |  

## Scenario 2: All Hotkeys = `move to next/previous hit`

🧮 **Expected Results**

ℹ️ Empty Text field = `navigate`

| Hotkey        | Live Preview | Reading | Source |
|:--------------|:-------------|:--------|:-------|
| CRTL+J/K      |  ✅          | ✅      | ✅     |
| CRTL+N/P      |  ✅          | ✅      | ✅     |
| Arrow UP/DOWN |  ✅          | ✅      | ✅     |
| TAB/Shift+TAB |  ✅          | ✅      | ✅     |  


ℹ️ Not Empty Text filed = `move to next/previous hit`

| Hotkey        | Live Preview | Reading | Source |
|:--------------|:-------------|:--------|:-------|
| CRTL+J/K      |  ✅          | ✅      | ✅     |
| CRTL+N/P      |  ✅          | ✅      | ✅     |
| Arrow UP/DOWN |  ✅          | ✅      | ✅     |
| TAB/Shift+TAB |  ✅          | ✅      | ✅     |

**Comment by tadashi-aikawa** [comment]
@FelipeRearden 
Wow! Professional work!! Thank you!! 😄 

One more piece of good news. I succeed to implement "Respect a header position both after opening dialog and after filtering" not only editor mode but **also reading mode** ✨

https://user-images.githubusercontent.com/9500018/167261044-40d1cfd5-9656-4765-8271-97cd615c5c61.mp4

**Comment by FelipeRearden** [comment]
> @FelipeRearden 
> 
> Wow! Professional work!! Thank you!! 😄 

Thank you very much @tadashi-aikawa !!!!! 💙 💙 💙

> One more piece of good news. I succeed to implement "Respect a header position both after opening dialog and after filtering" not only editor mode but **also reading mode** ✨

That's a huge and amazing good news !!!!!!!

Spectacular @tadashi-aikawa !!!!!! 

I must say that we have one of the most amazing feature of a plugin for Reading Mode related to the act of 📖 reading on Obsidian 👀

As someone who reads Obsidian notes that are long on Reading Mode with an iPadMini, I have to say that this is a ground breaking feature.

Thank you very much for always pursuit the limit to bring features that improves our Obsidian lives

It's an honor for me to participate with you and @chrisgrieser in this amazing FR 💙

**Comment by tadashi-aikawa** [comment]
@FelipeRearden 
Released [4.10.0-beta6](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.11.0-beta6) 🚀 

I wish it works on iPad 🙏

**Comment by FelipeRearden** [comment]
> Released [4.10.0-beta6](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.11.0-beta6) 🚀 

Starting tests now 🚀....

> I wish it works on iPad 🙏 

I'm gonna create an Vault test on iPad and we will find out 🙏

**Comment by FelipeRearden** [comment]
> > Released [4.10.0-beta6](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.11.0-beta6) 🚀 

One tiny detail about the new feature on `beta-6` only on **Reading Mode**

🖥 Obsidian 0.14.6 - My Long Note that I'm using to test 

What I'm seeing is this:

_I'm navigating using the mouse and I click under a header 1.1 for example_

--> **When I open the Modal, it is always ONE STEP ABOVE from where I am**

Example 1: 

├── 1. Heading [MODAL is HERE]
│ └── 1.1 Heading [I am HERE]
│   └──  1.1.1 heading


Example 2:

├── 1. Heading
│ └── 1.1 Heading [MODAL is HERE]
│   └──  1.1.1 heading [I am HERE]

Example 3:

├── 1. Heading
│ └── 1.1 Heading
│   └──  1.1.1 heading [MODAL is HERE]
├── 2. Heading [I am HERE]

---

It's very subtle and only occurs with mouse + click navigation 

If you cannot reproduce on your end, it's OKAY in my opinion :)

I'm just sharing because it was the operation that I do navigate:
1️⃣ scroll mouse to the end of the note
2️⃣ click mouse inside the header
3️⃣ open modal

Let's see the behavior on your end :)

**Comment by chrisgrieser** [comment]
> Released [4.10.0-beta6](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.11.0-beta6) 🚀

Works as intended – thank you! 🚀

**Comment by tadashi-aikawa** [comment]
Thanks @FelipeRearden :)

Unlike the editor mode, the concept of a cursor does not exist in the reading mode. Therefore AQS decides the current header from the scroll position.

In other words, "click mouse inside the header" affects nothing.

> --> When I open the Modal, it is always ONE STEP ABOVE from where I am

Perhaps, your "reading view" supposes to be like the one below.

![image](https://user-images.githubusercontent.com/9500018/167265495-96858268-78d7-4a57-9112-07cb277740f0.png)

**Comment by FelipeRearden** [comment]
> Perhaps, your "reading view" supposes to be like the one below.

Perfect @tadashi-aikawa !!! You nail it 🎯

Everything is perfect  with AQS my usage was biased when we have only one header in the top of the screen it's working perfect 

You are a legend -> It will took me 100 years to get this :)

---

All said, for me `beta-6` is working-effect here :)

**Comment by chrisgrieser** [comment]
I honestly can't think of anything else to improve. Think you can move this out of beta and close this issue! 🚀

**Comment by FelipeRearden** [comment]
> I honestly can't think of anything else to improve. Think you can move this out of beta and close this issue! 🚀 

@chrisgrieser I think @tadashi-aikawa will try to implement this idea that I had before the merge

> It's very simple to explain -> If a header is FOLD, when we select a Header from the Modal, it will open it !

**Comment by tadashi-aikawa** [comment]
> I'll try to implement your latest feature request in the next version 👍

Ah, sorry, `next version` is not `v6.11.0` but `v6.12.0`. So I'll release `v6.11.0` without your FR.

**Comment by tadashi-aikawa** [comment]
Released [v4.11.0](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.11.0) 🚀 

Thanks a lot!!

**Comment by FelipeRearden** [comment]
> Released [v4.11.0](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.11.0) 🚀 

Starting tests on iPadOS on version `4.11.1` right now 🚀

I get back to you in less than 30 minutes ....

**Comment by FelipeRearden** [comment]
> Ah, sorry, `next version` is not `v4.11.0` but `v4.12.0`. So I'll release `v4.11.0` without your FR.

Sorry @tadashi-aikawa for my misunderstanding :)

Great! I will Keep my eyes on the next version to do the tests :)

Let me know if you want me to open a new issue specially to this FR :)

**Comment by FelipeRearden** [comment]
> Released [v4.11.0](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.11.0) 🚀 

1️⃣ Support "reading view"
2️⃣ Eliminate unnecessary decorations as headers 
3️⃣ Respect a header position both after opening dialog and after filtering
4️⃣ Enable to change hotkeys about navigate and move to hit in the "header search in file" dialog
5️⃣ "move to next/previous hit" works just like "navigate" when the input query is empty
6️⃣ Fix a bug that doesn't scroll correctly on some files

✅ All Items are working perfect on version `4.11.1` on iPadOS 

> I wish it works on iPad 🙏

🎥 I think you gonna like to see this as much I did 💙
![video](https://user-images.githubusercontent.com/85180696/167292411-5ffd71d6-0ebe-43ce-83b3-96495672319b.MOV)

---

Thank you very much @tadashi-aikawa and @chrisgrieser  for this amazing new feature that will help me a lot :)

**Comment by tadashi-aikawa** [comment]
@FelipeRearden 
Thank you for your confirmation! It looks good 👍 

> Let me know if you want me to open a new issue specially to this FR :)
Yes, please 🙏

**Comment by FelipeRearden** [comment]
> @FelipeRearden Thank you for your confirmation! It looks good 👍

Thanks 😀

> > Let me know if you want me to open a new issue specially to this FR :)
> 
> Yes, please 🙏

Sure ! Done here in #42  

Have a great day @tadashi-aikawa ☀️

**Comment by tadashi-aikawa** [comment]
Next stage ➡️  https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/discussions/41
